### DISCUSSION TITLE
how to import buildings?

### DISCUSSION THREAD

**Discussion by foxneil0912** [osm2pgsql-dev/osm2pgsql#1576] [Q&A]
I want to convert addresses of buildings to point (longitude & latitude). How to create table with all addresses tags like addr:street, addr:housenumber and etc and to load data to this table? 
I tried to use command ` osm2pgsql central-fed-district-latest.osm.pbf -W -U postgres ` and got tables without addr:street (planet_osm_lines, planet_osm_point, planet_osm_polygon, planet_osm_roads)
Would be cool if someone would provide me lua script that creates one table with addresses and coordinates.

Thanks in advance!

**Comment by joto** [comment]
There is an example here how to import buildings: https://osm2pgsql.org/examples/buildings/ . You can then use PostGIS functions (`ST_Centroid` for instance) in your database to get a single point for each building.

**Reply by heckad** [reply]
But how to get full address? In example only buildings without full address. And after load we didn't find addr:street column. Is it correct?

**Reply by rustprooflabs** [reply]
I created a function to parse together the address so I can use `local address = get_address(object.tags)` when parsing buildings. 
 The `get_address()` function is defined in the [project's helpers.lua](https://github.com/rustprooflabs/pgosm-flex/blob/main/flex-config/helpers.lua#L384-L441) script.  Its awkwardly written but gets the job done.
